Pokemon/Pokemon/PKPokedexTableViewController.m

```diff
@@ -0,0 +1,481 @@
+                       @"magneton"
+                       ];
+    
+    
+    NSArray *water = @[
+                       @"blastoise",
+                       @"cloyster",
+                       @"dewgong",
+                       @"goldeen",
+                       @"golduck",
+                       @"gyarados",
+                       @"horsea",
+                       @"kabuto",
+                       @"kabutops",
+                       @"kingler",
+                       @"krabby",
+                       @"lapras",
+                       @"magikarp",
+                       @"omanyte",
+                       @"omastar",
+                       @"poliwag",
+                       @"poliwhirl",
+                       @"poliwrath",
+                       @"psyduck",
+                       @"seadra",
+                       @"seaking",
+                       @"seel",
+                       @"shellder",
+                       @"slowbro",
+                       @"slowpoke",
+                       @"squirtle",
+                       @"starmie",
+                       @"staryu",
+                       @"tentacool",
+                       @"tentacruel",
+                       @"vaporeon",
+                       @"wartortle"
+                       
+                       ];
+    
+    
+    
+    
+    self.pokemon = @{
+                     @"bug"      : bug,
+                     @"dragon"   : dragon,
+                     @"electric" : electric,
+                     @"fairy"    : fairy,
+                     @"fighting" : fighting,
+                     @"fire"     : fire,
+                     @"flying"   : flying,
+                     @"ghost"    :  ghost,
+                     @"grass"    : grass,
+                     @"ground"   : ground,
+                     @"ice"      : ice,
+                     @"normal"   : normal,
+                     @"poison"   : poison,
+                     @"psychic"  : psychic,
+                     @"rock"     : rock,
+                     @"steel"    : steel,
+                     @"water"    : water,
+                     };
+    
+    NSArray *keys = [self.pokemon allKeys];
+    NSString *key = keys[0];
+    NSArray *pokeArray = [self.pokemon objectForKey: key];
+    
+    self.allPokemonAtoZ = [[NSMutableArray alloc]init];
+    self.allPokemonAtoZ = [pokeArray mutableCopy];
+    
+    for (int i = 1; i < [keys count]; i++){
+        [self.allPokemonAtoZ addObjectsFromArray:[self.pokemon objectForKey:keys[i]]];
+    }
+    
+    NSOrderedSet *alphabetized = [NSOrderedSet orderedSetWithArray:self.allPokemonAtoZ];
+    self.allPokemonAtoZ = [alphabetized array];
+    self.allPokemonAtoZ = [self.allPokemonAtoZ sortedArrayUsingSelector:@selector(localizedCaseInsensitiveCompare:)];
+    
+}
+
+#pragma mark - Table view data source
+
+- (IBAction)segmentControlSelected:(UISegmentedControl *)sender {
+    if (self.segmentControl.selectedSegmentIndex == 0)
+    {
```

Comment:
very minor code styling thing - The '{' bracket should be at the end of the previous line, like this:
if (self.segmentControl.selectedSegmentIndex == 0) {

This is to just keep it consistent with the rest of your conditions for readability.